servers/src/main/java/tachyon/master/lineage/LineageMaster.java

```diff
@@ -177,17 +146,15 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
   public void start(boolean isLeader) throws IOException {
     super.start(isLeader);
     if (isLeader) {
-      mCheckpointExecutionService =
-          getExecutorService().submit(
-              new HeartbeatThread(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING,
-                  new CheckpointSchedulingExcecutor(this), mTachyonConf
-                      .getInt(Constants.MASTER_LINEAGE_CHECKPOINT_INTERVAL_MS)));
-      mRecomputeExecutionService =
-          getExecutorService().submit(
-              new HeartbeatThread(HeartbeatContext.MASTER_FILE_RECOMPUTATION,
-                  new RecomputeExecutor(new RecomputePlanner(mLineageStore, mFileSystemMaster),
-                      mFileSystemMaster), mTachyonConf
-                      .getInt(Constants.MASTER_LINEAGE_RECOMPUTE_INTERVAL_MS)));
+      mCheckpointExecutionService = getExecutorService()
```

Comment:
With this PR, the async persistence service is still quite coupled with Lineage: the periodic checkpoint-scheduling is started from here, and the only client of the checkpoint service is Lineage. Do you plan to expose this feature (directly or indirectly) to clients not using lineage as well?